jerry-core/ecma/builtin-objects/ecma-builtin-object.c

```diff
@@ -135,6 +135,141 @@ ecma_builtin_object_object_get_prototype_of (ecma_value_t this_arg, /**< 'this'
   return ret_value;
 } /* ecma_builtin_object_object_get_prototype_of */
 
+#ifndef CONFIG_DISABLE_ES2015_BUILTIN
+/**
+ * [[SetPrototypeOf]]
+ *
+ * See also:
+ *          ES2015 9.1.2
+ */
+static bool
+ecma_set_prototype_of (ecma_value_t o_value, /**< O */
+                       ecma_value_t v_value) /**< V */
+{
+  /* 1. */
+  JERRY_ASSERT (ecma_is_value_object (o_value));
+  JERRY_ASSERT (ecma_is_value_object (v_value) || ecma_is_value_null (v_value));
+
+  ecma_object_t *o_p = ecma_get_object_from_value (o_value);
+
+  /* 3. */
+  ecma_object_t *current_p = ecma_get_object_prototype (o_p);
+  ecma_value_t current_value = ecma_make_object_value (current_p);
+
+  /* 4. */
+  if (ecma_op_same_value (v_value, current_value))
+  {
+    return true;
+  }
+
+  /* 2., 5. */
+  if (!ecma_get_object_extensible (o_p))
+  {
+    return false;
+  }
+
+  if (!ecma_is_value_null (v_value))
+  {
+    ecma_object_t *v_p = ecma_get_object_from_value (v_value);
+
+    /* 6., 7., 8. */
+    ecma_object_t *p_p = v_p;
+    while (true)
+    {
+      /* a. */
+      if (p_p == NULL)
+      {
+        break;
+      }
+
+      /* b. */
+      ecma_value_t p_value = ecma_make_object_value (p_p);
+      if (ecma_op_same_value (p_value, o_value))
+      {
+        return false;
+      }
+
+      /* c.i. TODO: es2015-subset profile does not support having a different
+       * [[GetPrototypeOf]] internal method */
```

Comment:
I think the spec means "if the object has its own specific [[getPrototypeOf]] method, we can break the loop", rather than "does not support "